frontend/covid-dashboard-frontend/src/components/HomePage.js

```diff
@@ -37,155 +35,185 @@ const HomePage = ({cities, states, countries, loading, cityDetail, stateDetail,
+  const chartableData = (type, data) => {
+    const series = [{
       name: type,
       type: 'line',
-      data: data.map(item => item[type])
-    };
+      data: data.map(item => {
+        return [DateTime.fromHTTP(item.date).toISODate(), item[type]]
+      })
+    }];
 
     const xAxis = {
-      type: 'category',
-      data: data.map(item => item.date)
+      type: 'time',
+      splitLine: { show: false }
     }
 
     const yAxis = {
       type: 'value'
     }
 
+    const dataZoom = [{
+      type: 'slider',
+      xAxisIndex: 0,
+      filterMode: 'empty'
+    }]
+
     const option = {
-      toolbox: commonToolBox(),
+      toolbox: {
+        feature: {
+          saveAsImage: { title: 'Save As Image' },
+          dataView: { readOnly: false, title: 'Data', lang: ['', 'Close', 'Refresh'] }
+        }
+      },
       tooltip: {
         trigger: 'item',
-        formatter: (params) => `Date: ${params.name}<br>Name: ${params.seriesName}<br>Total: ${params.data}`
+        formatter: (params) => `Name: ${_.startCase(params.seriesName)}<br>Date: ${params.data[0]}<br>Total: ${params.data[1]}`
       },
+      dataZoom,
       xAxis,
       yAxis,
-      series: [series]
+      series
     }
-
     return option;
   }
 
-  const chartableData = () => {
-    if (locationType === 'city') {
-      return cityDetail;
-    } else if (locationType === 'state') {
-      return stateDetail;
+  const locationData = () => {
+    let data;
+    if (locationType === CITY_LOCATION) {
+      data = cityDetail;
+    } else if (locationType === STATE_LOCATION) {
+      data = stateDetail;
+    } else if (locationType === COUNTRY_LOCATION) {
+      data = countryDetail;
     }
-    return countryDetail;
+    return data;
   }
 
   const refreshCharts = () => {
-    setIsReady(false);
-    setCasesOptions(makeData('cases', chartableData(), start, end));
-    setDeathsOptions(makeData('deaths', chartableData(), start, end));
-    setRecoveriesOptions(makeData('recoveries', chartableData(), start, end));
-    setHospitalizationRateOptions(makeData('hospitalization_rate', chartableData(), start, end));
-    setTestingRateOptions(makeData('testing_rate', chartableData(), start, end));
-    setIsReady(true);
+    const locData = locationData();
+
+    if (loading || _.isEmpty(locData)) {
+      clearChartData();
+      return;
+    }
+
+    // probably makes decent sense to filter by date here
+    const filteredData = locData.filter(item => true)
+
+    setCasesOptions(chartableData('cases', filteredData));
+    setDeathsOptions(chartableData('deaths', filteredData));
+    setRecoveriesOptions(chartableData('recoveries', filteredData));
+    setHospitalizationRateOptions(chartableData('hospitalization_rate', filteredData));
+    setTestingRateOptions(chartableData('testing_rate', filteredData));
   };
 
   const loadOptions = (inputValue, callback) => {
-    // _.delay((q) => {
-      if (!inputValue) {
+    _.delay((txt) => {
+      if (!txt) {
```

Comment:
added back in lodash delay to debounce the input change events when typing into the location search bar